doc/sphinx/how-to/how-to-test-mir-for-a-release.md

```diff
@@ -0,0 +1,174 @@
+
+1. Test that the following QT Wayland applications can be started and can receive input:
+    ```sh
+    sudo apt install qtwayland5 kate qterminal
+    
+    # First...
+    kate
+   
+    # Then...
+    qtwayland5
+   
+    # Finally..
+    qterminal
+    ```
+2. Test that `weston-terminal` can be started and can receive input:
+    ```
+    sudo apt install weston
+    weston
+    ```
+3. Test that `glmark2-wayland` can be run:
+    ```sh
+    sudo apt install glmark2-wayland
+    glmark2-wayland
+    ```
+4. Test that `gnome-terminal` can be started and can receive input:
+   ```sh
+   sudo apt install gnome-terminal
+   gnome-terminal
+   ```
+
+### Mir Configuration Options
+
+1. Run any Mir compositor with different console providers and ensure that the compositor can start:
+    ```sh
+    miral-app --console-provider=vt --vt=4
+    ```
+    - This requires running with root privileges 
+    - You need to ensure that `XDG_RUNTIME_DIR` is set in the environment. If using `sudo`, 
+      it might strip this out; running something like `sudo env XDG_RUNTIME_DIR=/run/user/1000 miral-shell ...`
+      will ensure this is set. 
+    
+    ```sh
+    miral-app --console-provider=logind
+    ```
+    - You can switch to vt4 and sign in
+    
+    ```sh
+    miral-app --console-provider=minimal 
+    ```
+    - This is used when all others fail
+    - This does not provide VT switching capabilities (Ctrl-Alt-F1, etc) 
+    - This is _only_ used for the `gbm-x11`, `gbm-wayland`, and `virtual` platforms 
+
+2. Run `miral-app` with different window managers and confirm that the window management
+   styles work as expected:
+    ```sh
+    miral-app --window-manager=floating # traditional floating window manager
+    miral-app --window-manager=tiling # a tiling window manager
+    miral-app --kiosk # a typical kiosk
+    ```
+
+### Run Compositors that are Built for Testing
+1. Run the `mir-test-smoke-runner` and confirm that the final output is:
+   `Smoke testing complete with returncode 0`.
+2. Run the tests found in `mir-test-tools` and confirm the contents on the screen. A useful script for doing this is:
+   ```bash
+    #! /bin/bash
+
+    set -e
+   
+    if [ -x "$(command -v fgconsole)" ]
+    then
+        trap "sudo chvt $(sudo fgconsole)" EXIT
+    fi
+   
+    if [ -v DISPLAY ]
+    then
+        snap run mir-test-tools.gtk3-test
+        snap run mir-test-tools.qt-test
+        snap run mir-test-tools.sdl2-test
+        snap run mir-test-tools.smoke-test
+        snap run mir-test-tools.performance-test
+        snap run mir-test-tools.mir-flutter-app
+    fi
+   
+    sudo snap run mir-test-tools.gtk3-test
+    sudo snap run mir-test-tools.qt-test
+    sudo snap run mir-test-tools.sdl2-test
+    sudo snap run mir-test-tools.smoke-test
+    sudo snap run mir-test-tools.performance-test
+    sudo snap run mir-test-tools.mir-flutter-app
+   
+    echo All tests passed
+   ```
```

Comment:
We really need a better place for this script (maybe it belongs in the snap and run by `/snap/mir-test-tools/current/selftest.sh`) but that's pre-existing.

Reply:
That sounds better (note it needs running outside the snap environment)

Reply:
How do I make it so that it can run outside of the snap environment in a snap? Wrap it in something like https://github.com/Miriway/Miriway/blob/main/miriway-unsnap?

Reply:
No, you need to have the user execute `/snap/mir-test-tools/current/…`, rather than a `mir-test-tools.*` app.

Reply:
Ah I see what you mean! I'll try that

Reply:
Cool :+1: